pkg/image/importer/importer.go

```diff
@@ -509,6 +510,14 @@ func (isi *ImageStreamImporter) importRepositoryFromDocker(ctx gocontext.Context
 				continue
 			}
 		}
+
+		signatures, err := signature.GetSignatureForImage(repository.Registry.Hostname(), repository.Name, importDigest.Image.Name)
+		if err == nil {
+			importDigest.Image.Signatures = signatures
```

Comment:
Does this overwrite all pre-existing signatures?

What happens if the same image (same manifest digest) is published in two different repositories, with two different signatures? IIRC the signatures are attached to the unique image object, and shared between all image streams; so the imports should be adding signatures (removing duplicates), not removing the old ones.

Reply:
Is this going to be addressed?

Reply:
It seems likely that we may want a backfill controller that keeps registry signatures up to date.  

That may be a lot like a mirroring controller that ensures that images are kept in sync.

Reply:
Maybe we could re-use the scheduled importer to do just that?